### PR TITLE
[SC-906] Refactor/power pods tests

### PR DESCRIPTION
**Pull request by SevenSwen**

### CODE REVIEW

File: test/PowerPod.js [hunk 1]
```diff
@@ -1,60 +1,154 @@
...
-        await delegation.connect(user).delegate(user.address);
-    };
-
     async function initContracts() {
-        const TokenPermitMock = await ethers.getContractFactory('ERC20PermitMock');
-        const oneInch = await TokenPermitMock.deploy('1inch', '1inch', addr.address, ether('200'));
-        await oneInch.deployed();
+        const commonLockDuration = time.duration.days('40');
+        const MAX_WHITELISTED = 3;
+        const BALANCE_THRESHOLD = 1000; // 10%
```
Comment (ZumZoom) [inline test/PowerPod.js:10]: Constants should be somewhere on top of the file.

Also `commonLockDuration => COMMON_LOCK_DURATION`.

File: test/PowerPod.js [hunk 2]
```diff
@@ -1,60 +1,154 @@
...
+                    `A${userIndex}DT`,
+                );
+            await delegation.connect(user).delegate(user.address);
+        };
+
+        const depositAndDelegateTo = async (st1inch, delegation, from, to, amount, duration = commonLockDuration) => {
+            await st1inch.connect(from).deposit(amount, duration);
+            await st1inch.connect(from).addPod(delegation.address);
+            await delegation.connect(from).delegate(to);
+        };
```
Comment (ZumZoom) [inline test/PowerPod.js:29]: Should be declared somewhere on top of the file like `async function depositAndDelegateTo(st1inch, delegation, from, to, amount, duration = commonLockDuration)`

File: test/PowerPod.js [hunk 3]
```diff
@@ -1,60 +1,154 @@
...
+        await stakeAndRegisterInDelegation(st1inch, delegation, owner, ether('1'), 0);
 
-    before(async function () {
-        accounts = await ethers.getSigners();
-        addr = accounts[0];
-        addr1 = accounts[1];
-    });
+        return {
+            contracts: { st1inch, delegation, whitelistRegistry },
+            accounts: { owner, alice, accounts },
```
Comment (ZumZoom) [inline test/PowerPod.js:55]: That's too much, if we unwrap accounts, we can return only the unwrapped ones like `accounts: { owner, alice, bob, ... }`

File: test/PowerPod.js [hunk 4]
```diff
@@ -1,60 +1,154 @@
...
+            other: { commonLockDuration },
+        };
+    }
 
     describe('Should calculate voting power', function () {
         it('for account with 0 balance', async function () {
-            const { delegation } = await loadFixture(initContracts);
-            expect(await delegation.votingPowerOf(addr.address)).to.equal(0);
+            const { contracts: { delegation }, accounts: { accounts } } = await loadFixture(initContracts);
+            expect(await delegation.votingPowerOf(accounts[9].address)).to.equal(0);
```
Comment (ZumZoom) [inline test/PowerPod.js:64]: Why `accounts[9]`?

**Review by ZumZoom — CHANGES_REQUESTED** [review]

File: test/PowerPod.js (hunk 1 above)
Comment (SevenSwen) [reply test/PowerPod.js:10]: fixed

File: test/PowerPod.js (hunk 2 above)
Comment (SevenSwen) [reply test/PowerPod.js:29]: fixed

File: test/PowerPod.js (hunk 4 above)
Comment (SevenSwen) [reply test/PowerPod.js:64]: Because we need clear address without some balances. We can use anything address after fifth.

Comment (SevenSwen) [reply test/PowerPod.js]: renamed

**Review by bitcoinfinancier — APPROVED** [review]

**Review by ZumZoom — APPROVED** [review]

**Pull request merged**